sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SqaleXmlLoader.java

```diff
@@ -0,0 +1,165 @@
+      xmlFactory.setProperty(XMLInputFactory.IS_COALESCING, Boolean.TRUE);
+      xmlFactory.setProperty(XMLInputFactory.IS_NAMESPACE_AWARE, Boolean.FALSE);
+      // just so it won't try to load DTD in if there's DOCTYPE
+      xmlFactory.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
+      xmlFactory.setProperty(XMLInputFactory.IS_VALIDATING, Boolean.FALSE);
+      SMInputFactory inputFactory = new SMInputFactory(xmlFactory);
+      try {
+        processRoot(reader, inputFactory);
+      } catch (XMLStreamException e) {
+        throw new IllegalStateException("SQALE XML is not valid: " + resourcePath, e);
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Could not read " + resourcePath, e);
+    }
+  }
+
+  private void processRoot(Reader reader, SMInputFactory inputFactory) throws XMLStreamException {
+    SMHierarchicCursor rootCursor = inputFactory.rootElementCursor(reader);
+    rootCursor.advance();
+    SMInputCursor charCursor = rootCursor.childElementCursor("chc");
+    while (charCursor.getNext() != null) {
+      SMInputCursor subCharCursor = charCursor.childElementCursor("chc");
+      while (subCharCursor.getNext() != null) {
+        processSubChar(subCharCursor);
+      }
+    }
+  }
+
+  private void processSubChar(SMInputCursor subCharCursor) throws XMLStreamException {
+    String subCharName = null;
+    SMInputCursor subCharChildCursor = subCharCursor.childElementCursor();
+    while (subCharChildCursor.getNext() != null) {
+      String childName = subCharChildCursor.getLocalName();
+      if ("key".equals(childName)) {
+        subCharName = subCharChildCursor.getElemStringValue();
+      }
+      if ("chc".equals(childName)) {
+        processRule(subCharName, subCharChildCursor);
+      }
+    }
+  }
+
+  private void processRule(String subCharName, SMInputCursor ruleCursor) throws XMLStreamException {
+    SMInputCursor ruleChildCursor = ruleCursor.childElementCursor();
+    String ruleKey = null;
+    String remediationFunction = null;
+    String remediationFactor = null;
+    String offset = null;
+    while (ruleChildCursor.getNext() != null) {
+      String childName = ruleChildCursor.getLocalName();
+      if ("rule-key".equals(childName)) {
+        ruleKey = ruleChildCursor.getElemStringValue();
+      }
+      if ("prop".equals(childName)) {
+        Map<String, String> propMap = childrenMap(ruleChildCursor.childElementCursor());
+        String key = propMap.get("key");
+        if ("remediationFunction".equals(key)) {
+          remediationFunction = propMap.get("txt");
+        }
+        if ("offset".equals(key)) {
+          offset = timeValue(propMap);
+        }
+        if ("remediationFactor".equals(key)) {
+          remediationFactor = timeValue(propMap);
+        }
+      }
+    }
+    NewRule rule = repository.rule(ruleKey);
+    if (rule != null) {
+      rule.setDebtSubCharacteristic(subCharName);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "setDebtSubCharacteristic"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)